constraint/config/crds/externaldata.gatekeeper.sh_providers.yaml

```diff
@@ -54,10 +57,44 @@ spec:
             type: object
         type: object
     served: true
+    storage: false
+  - name: v1beta1
+    schema:
+      openAPIV3Schema:
+        description: Provider is the Schema for the providers API
+        properties:
+          apiVersion:
+            description: 'APIVersion defines the versioned schema of this representation
+              of an object. Servers should convert recognized schemas to the latest
+              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            type: string
+          kind:
+            description: 'Kind is a string value representing the REST resource this
+              object represents. Servers may infer this from the endpoint the client
+              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: Spec defines the Provider specifications.
+            properties:
+              caBundle:
+                description: CABundle is a base64-encoded string that contains the
+                  TLS CA bundle in PEM format. It is used to verify the signature
+                  of the provider's certificate.
+                type: string
+              insecureTLSSkipVerify:
```

Comment:
we can still require `https` and deprecate this field.

Reply:
We actually can't per K8s backwards compatibility guidelines:

This would be a behavioral change:

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#on-compatibility

And v1beta1 signals backwards compatibility:

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#incompatible-api-changes

After we move to v1beta1, we can no longer deprecate the field without risking breaking users who depend on it. I can see exceptions for unforeseen circumstances or security vulnerabilities, but that's not the case here.

TL;DR once we go to beta, if this field exists, it's here to stay. Personally, I'm fine with that, but it's worth highlighting.

Reply:
Why not simply drop this field?  Require `https` and support `http` on `::1` and `127.0.0.1`.

Reply:
Removed `insecureTLSSkipVerify` for now.